tests/unit/lms/views/basic_lti_launch_test.py

```diff
@@ -430,6 +433,20 @@ def test_it_adds_the_document_url(self, context, pyramid_request):
             pyramid_request.parsed_params["document_url"]
         )
 
+    def test_it_saves_the_group_set_id_to_the_db(
+        self, assignment_service, context, pyramid_request
+    ):
+        configure_assignment_caller(
+            context, pyramid_request, parsed_params={"group_set": "ID"}
+        )
+
+        assignment_service.upsert.assert_called_once_with(
+            pyramid_request.parsed_params["document_url"],
+            pyramid_request.parsed_params["tool_consumer_instance_guid"],
+            pyramid_request.parsed_params["resource_link_id"],
+            extra={"group_set_id": "ID"},
+        )
```

Comment:
I think it might be better to integrate this test into the [`test_it_saves_the_assignments_document_url_to_the_db()`](https://github.com/hypothesis/lms/blob/13cc730be29007f8de041499242e114ef1899b12/tests/unit/lms/views/basic_lti_launch_test.py#L412-L422) test above. And to be honest the two other tests in this class could also just be two assertions added to a single test:

```python
class TestConfigureAssignment:
    @pytest.mark.parametrize(
        "parsed_params,expected_extras",
        [
            (None, {}),
            ({"group_set": 42}, {"group_set_id": 42}),
        ],
    )
    def test_it(
        self,
        assignment_service,
        context,
        pyramid_request,
        parsed_params,
        expected_extras,
    ):
        configure_assignment_caller(context, pyramid_request, parsed_params)

        assignment_service.upsert.assert_called_once_with(
            pyramid_request.parsed_params["document_url"],
            pyramid_request.parsed_params["tool_consumer_instance_guid"],
            pyramid_request.parsed_params["resource_link_id"],
            extra=expected_extras,
        )
        context.js_config.add_document_url.assert_called_once_with(
            pyramid_request.parsed_params["document_url"]
        )
        context.js_config.maybe_enable_grading.assert_called_once_with()
```